tests/test_adat.py

```diff
@@ -46,6 +69,87 @@ def adat_duration(level, partial):
         duration = 10
     return duration
 
+@pytest.mark.uncollect_if(func=adat_input_uncollect)
+@pytest.mark.parametrize(["board", "config"], list_configs())
+def test_adat_input(pytestconfig, board, config):
+    features = get_config_features(board, config)
+
+    adapter_dut, adapter_harness = get_xtag_dut_and_harness(pytestconfig, board)
+    duration = adat_duration(pytestconfig.getoption("level"), features["partial"])
+    fail_str = ""
+
+    with XrunDut(adapter_dut, board, config) as dut:
+        for fs in features["samp_freqs"]:
+            assert features["analogue_i"] == 8
+            if fs <= 48000:
+                num_in_channels = 16
+            elif fs <= 96000:
+                num_in_channels = 12
+            else:
+                num_in_channels = 10
```

Comment:
Optional comment: I think it would make more sense to set num_dig_in_channels to 8, 4 or 2 in these ifs, and then add to features["analogue_i"] below to get num_in_channels. It gets all the same results, but I feel that makes better sense logically as the number of digital channels is actually what is determined by the sample rate.